src/libraries/System.Text.Json/src/System/Text/Json/Serialization/Metadata/JsonPropertyInfoOfT.cs

```diff
@@ -408,13 +411,25 @@ internal override bool ReadJsonAsObject(ref ReadStack state, ref Utf8JsonReader
             return success;
         }
 
-        internal override void SetExtensionDictionaryAsObject(object obj, object? extensionDict)
+        internal sealed override void SetExtensionDictionaryAsObject(object obj, object? extensionDict)
         {
             Debug.Assert(HasSetter);
             T typedValue = (T)extensionDict!;
             Set!(obj, typedValue);
         }
 
+        internal sealed override void SetValueAsObject(object obj, object? value, ref ReadStack state)
```

Comment:
Same feedback as https://github.com/dotnet/runtime/pull/72937/files#r931338660 seems to apply here. Is this method needed given that we have the untyped `Set` delegate?